pyanaconda/modules/boss/installation.py

```diff
@@ -0,0 +1,151 @@
+        log.info("Copying logs from the installation environment...")
+
+        if flags.nosave_logs:
+            log.warning("Installation logs will not be saved to the installed system due to the "
+                        "nosave option.")
+        else:
+            self._create_logs_directory()
+            self._copy_tmp_logs()
+            self._copy_pre_script_logs()
+            self._copy_dnf_debugdata()
+            self._copy_post_script_logs()
+            self._dump_journal()
+            self._chmod_logs()
+
+            log.info("Done.")
+
+    def _create_logs_directory(self):
+        """Create directory for Anaconda logs on the install target"""
+        util.mkdirChain(self._sysroot + ANACONDA_LOG_DIR)
+
+    def _copy_tmp_logs(self):
+        """Copy a number of log files from tmp"""
+        LOG_FILES_TO_COPY = [
+            "anaconda.log", "syslog", "X.log", "program.log", "packaging.log", "storage.log",
+            "ifcfg.log", "lvm.log", "dnf.librepo.log", "hawkey.log", "dbus.log"
+        ]
+        for logfile in LOG_FILES_TO_COPY:
+            self._copy_file_to_sysroot("/tmp/" + logfile,
+                                       ANACONDA_LOG_DIR + logfile)
+        # copy also list of packages used for creating the installation media
+        self._copy_file_to_sysroot("/root/lorax-packages.log",
+                                   ANACONDA_LOG_DIR + "lorax-packages.log")
+
+    def _copy_pre_script_logs(self):
+        """Copy logs from %pre scripts"""
+        self._copy_tree_to_sysroot("/tmp/pre-anaconda-logs",
+                                   ANACONDA_LOG_DIR)
+
+    def _copy_dnf_debugdata(self):
+        """Copy DNF debug data"""
+        self._copy_tree_to_sysroot("/root/debugdata",
+                                   ANACONDA_LOG_DIR + "dnf_debugdata/")
+
+    def _copy_post_script_logs(self):
+        """Copy logs from %post scripts"""
+        for logfile in glob.glob("/tmp/ks-script*.log"):
+            self._copy_file_to_sysroot(logfile,
+                                       ANACONDA_LOG_DIR + os.path.basename(logfile))
+
+    def _dump_journal(self):
+        """Dump journal from the installation environment"""
+        with open(self._sysroot + ANACONDA_LOG_DIR + "journal.log", "w") as logfile:
+            util.execWithRedirect("journalctl", ["-b"], stdout=logfile)
+
+    def _chmod_logs(self):
+        """Set access bits"""
+        util.execInSysroot("bash", ["-c", "chmod 0600 " + ANACONDA_LOG_DIR + "*"])
+
+    def _copy_kickstart(self):
+        """Copy input kickstart file"""
+        log.info("Copying input kickstart file...")
+
+        if flags.nosave_input_ks:
+            log.warning("Input kickstart will not be saved to the installed system due to the "
+                        "nosave option.")
+        else:
+            self._copy_file_to_sysroot("/run/install/ks.cfg",
+                                       "/root/original-ks.cfg")
+            log.info("Done.")
```

Comment:
I am not sure if the `Done.` messages are necessary. How the logs look like with them?